### ISSUE TITLE
Restart training from presaved checkpoint doesn't seem to work

### ISSUE THREAD

**Issue by vijay-yajiv** [ultralytics/yolov5#316]
## 🐛 Bug
I want to train my Yolov5 Large model for 100 epochs - but this takes a lot of time.
So I plan to split it into 3 parts of 30,30,40 epochs and continue training from last checkpoint by passing the --weights parameter.

**Part1**:  I train my data for 30 epochs and save the last.pt at the end of training. My results at the end of this training are 0.03 GIoU and 0.94 MAP

**Part2**: I continue training for the next 30 epochs by passing 
--epochs 100 and --weights last.pt

However, when I do the above, in the 2nd phase of training, _all the metrics drastically fall as if the training were restarting._  I would expect the metrics to start with somewhat similiar numbers to what were present at the end of 30 Epochs...but the metrics seem to be as if the training is re-starting from a scratch.

Is this a possible bug or am I doing something wrong.
In the train.py there is something like "Saving weights after stripping optimizer" - Does this have any role to play ?

Thank you.

Note: I have seen that you suggest avoiding --resume, so I have not used that and gone down the route of providing --weights parameter

**Comment by NTU-P04922004** [comment]
This seems to be the learning rate reset problem like this issue:
https://github.com/ultralytics/yolov5/issues/292

Have you tried this fix?
https://github.com/ultralytics/yolov5/pull/300

**Comment by vijay-yajiv** [comment]
No I haven't  - I could try it, although at a glance, I didn't get too much of an idea of how to solve it.
Can you guide a bit please. Thanks.

**Comment by glenn-jocher** [comment]
@NTU-P04922004 @vijay-yajiv this is definitely not how to train. There is no such thing as 'breaking up training into parts'. Train fully from start to finish.
